shell/platform/android/test/io/flutter/plugin/platform/PlatformPluginTest.java

```diff
@@ -325,48 +327,99 @@ public void setSystemUiMode() {
+      verify(fakeDecorView).setSystemUiVisibility(View.SYSTEM_UI_FLAG_HIDE_NAVIGATION);
       verify(fakeDecorView)
           .setSystemUiVisibility(
               View.SYSTEM_UI_FLAG_LAYOUT_STABLE
                   | View.SYSTEM_UI_FLAG_LAYOUT_HIDE_NAVIGATION
-                  | View.SYSTEM_UI_FLAG_LAYOUT_FULLSCREEN
-                  | View.SYSTEM_UI_FLAG_HIDE_NAVIGATION
-                  | View.SYSTEM_UI_FLAG_FULLSCREEN);
+                  | View.SYSTEM_UI_FLAG_LAYOUT_FULLSCREEN);
 
       platformPlugin.mPlatformMessageHandler.showSystemUiMode(
           PlatformChannel.SystemUiMode.IMMERSIVE);
-      verify(fakeDecorView)
+
+      verify(fakeDecorView).setSystemUiVisibility(View.SYSTEM_UI_FLAG_IMMERSIVE);
+      verify(fakeDecorView, times(2)).setSystemUiVisibility(View.SYSTEM_UI_FLAG_FULLSCREEN);
+      verify(fakeDecorView, times(2)).setSystemUiVisibility(View.SYSTEM_UI_FLAG_HIDE_NAVIGATION);
+      verify(fakeDecorView, times(2))
           .setSystemUiVisibility(
-              View.SYSTEM_UI_FLAG_IMMERSIVE
-                  | View.SYSTEM_UI_FLAG_LAYOUT_STABLE
+              View.SYSTEM_UI_FLAG_LAYOUT_STABLE
                   | View.SYSTEM_UI_FLAG_LAYOUT_HIDE_NAVIGATION
-                  | View.SYSTEM_UI_FLAG_LAYOUT_FULLSCREEN
-                  | View.SYSTEM_UI_FLAG_HIDE_NAVIGATION
-                  | View.SYSTEM_UI_FLAG_FULLSCREEN);
+                  | View.SYSTEM_UI_FLAG_LAYOUT_FULLSCREEN);
 
       platformPlugin.mPlatformMessageHandler.showSystemUiMode(
           PlatformChannel.SystemUiMode.IMMERSIVE_STICKY);
-      verify(fakeDecorView)
+
+      verify(fakeDecorView).setSystemUiVisibility(View.SYSTEM_UI_FLAG_IMMERSIVE_STICKY);
+      verify(fakeDecorView, times(3)).setSystemUiVisibility(View.SYSTEM_UI_FLAG_FULLSCREEN);
+      verify(fakeDecorView, times(3)).setSystemUiVisibility(View.SYSTEM_UI_FLAG_HIDE_NAVIGATION);
+      verify(fakeDecorView, times(3))
           .setSystemUiVisibility(
-              View.SYSTEM_UI_FLAG_IMMERSIVE_STICKY
-                  | View.SYSTEM_UI_FLAG_LAYOUT_STABLE
+              View.SYSTEM_UI_FLAG_LAYOUT_STABLE
                   | View.SYSTEM_UI_FLAG_LAYOUT_HIDE_NAVIGATION
-                  | View.SYSTEM_UI_FLAG_LAYOUT_FULLSCREEN
-                  | View.SYSTEM_UI_FLAG_HIDE_NAVIGATION
-                  | View.SYSTEM_UI_FLAG_FULLSCREEN);
+                  | View.SYSTEM_UI_FLAG_LAYOUT_FULLSCREEN);
     }
 
     if (Build.VERSION.SDK_INT >= 29) {
       platformPlugin.mPlatformMessageHandler.showSystemUiMode(
           PlatformChannel.SystemUiMode.EDGE_TO_EDGE);
-      verify(fakeDecorView)
+
+      verify(fakeDecorView, times(4))
           .setSystemUiVisibility(
               View.SYSTEM_UI_FLAG_LAYOUT_STABLE
                   | View.SYSTEM_UI_FLAG_LAYOUT_HIDE_NAVIGATION
                   | View.SYSTEM_UI_FLAG_LAYOUT_FULLSCREEN);
     }
   }
 
+  @Config(minSdk = 30)
+  @Test
+  public void setSystemUiMode() {
+    if (Build.VERSION.SDK_INT >= 30) {
```

Comment:
nit: I don't think this condition is required

Reply:
I kept getting this error without it:
```
../../../flutter/shell/platform/android/test/io/flutter/plugin/platform/PlatformPluginTest.java:387: Error: Class requires API level 30 (current min is 16): android.view.WindowInsetsController [NewApi]
    fakeWindowInsetsController = mock(WindowInsetsController.class);
                                      ~~~~~~~~~~~~~~~~~~~~~~
../../../flutter/shell/platform/android/test/io/flutter/plugin/platform/PlatformPluginTest.java:388: Error: Call requires API level 30 (current min is 16): android.view.Window#getInsetsController [NewApi]
```
Is there a better way to get around this?